src/main/java/one/util/streamex/AbstractStreamEx.java

```diff
@@ -252,6 +254,61 @@ public S skip(long n) {
         return supply(stream().skip(n));
     }
 
+    /**
+     * Returns a new, non-splittable, sequential stream containing
+     * all the elements of the original stream except the last n elements.
+     * Consumes space proportional to the number of elements skipped.
+     *
+     * <p>
+     * For example, {@code StreamEx.of("a", "b", "c").skipLast(1)} will yield a stream containing
+     * two elements: a, b.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">quasi-intermediate operation</a>.
+     *
+     * @param n the non-negative number of elements to leave off the end of the stream.
+     * @return the new, sequential stream
+     * @throws IllegalArgumentException if {@code n} is negative or {@link Integer#MAX_VALUE}
+     * @since 0.6.7
+     */
+    public S skipLast(int n) {
+        if (n < 0 || n == Integer.MAX_VALUE)
+            throw new IllegalArgumentException(Long.toString(n));
+        if (n == 0)
+            return supply(this);
+
+        BlockingDeque<T> buffer = new LinkedBlockingDeque<>(n + 1);
```

Comment:
Note that an optimized solution could be provided if the stream is SIZED/sequential or SIZED/SUBSIZED/parallel. In this case you don't need buffering at all as you know exactly how many elements are left. Also why concurrent buffer if you just refuse to parallelize? ArrayDeque (or simply Object[] array) would suffice and would be much faster.